swagger_spec_validator/ref_validators.py

```diff
@@ -157,7 +157,7 @@ def deref_and_validate(validator, schema_element, instance, schema,
         the swagger service spec.
     :param default_validator_callable: jsonschema._validators.* callable
     """
-    if isinstance(instance, dict) and '$ref' in instance:
+    if isinstance(instance, dict) and '$ref' in instance and isinstance(instance['$ref'], str):
```

Comment:
``six.string_types`` should solve your issues around checking if an object is a string

Reply:
For some reason I didn't realize you already had six as a dependency.  I'm using six for the type check now.